zha/zigbee/cluster_handlers/manufacturerspecific.py

```diff
@@ -496,3 +499,64 @@ class DanfossDiagnosticClusterHandler(DiagnosticClusterHandler):
         AttrReportConfig(attr="sw_error_code", config=REPORT_CONFIG_DEFAULT),
         AttrReportConfig(attr="motor_step_counter", config=REPORT_CONFIG_DEFAULT),
     )
+
+
+@registries.CLUSTER_HANDLER_ONLY_CLUSTERS.register(SINOPE_MANUFACTURER_CLUSTER)
+@registries.CLUSTER_HANDLER_REGISTRY.register(SINOPE_MANUFACTURER_CLUSTER)
+class SinopeManufacturerClusterHandler(ClusterHandler):
+    """Sinope Manufacturer cluster handler."""
+
+    BIND = True
+
+    def __init__(self, cluster: zigpy.zcl.Cluster, endpoint: Endpoint) -> None:
+        """Initialize Sinope cluster handler."""
+        super().__init__(cluster, endpoint)
+        shared_attr = {
+            "double_up_full": True,
+            "on_led_color": True,
+            "off_led_color": True,
+            "off_led_intensity": True,
+            "on_led_intensity": True,
+        }
+        if self.cluster.endpoint.model in [
+            "DM2550ZB",
+            "DM2550ZB-G2",
+            "DM2500ZB-G2",
+            "DM2500ZB",
+        ]:
+            self.ZCL_INIT_ATTRS = {
+                **shared_attr,
+                "on_intensity": True,
+            }
+
+        elif self.cluster.endpoint.model in ["SW2500ZB", "SW2500ZB-G2"]:
+            self.ZCL_INIT_ATTRS = shared_attr.copy()
+
+    REPORT_CONFIG = (
+        AttrReportConfig(
+            attr="action_report",
+            config=(
+                REPORT_CONFIG_MIN_INT_IMMEDIATE,
+                REPORT_CONFIG_MIN_INT_IMMEDIATE,
```

Comment:
Do we really want 0 for min and max?

Reply:
I'm fairly new to the Zigbee spec so I probably got this wrong. 

My understanding is that it specifies how frequently the device should send updates about the `action_report` attribute, which for Sinope switches is how it communicates any button presses on the switch, especially non-single-press, like "long press" or "double tap".  

I figured 0 made sense because it'll be responsive when you press the switch. Do you think this should be a larger value on the Max side?

Reply:
the maximum interval is for sending a report even if the value hasn't changed. The way this is configured at the moment fits this case I believe:

"If this value is set to 0x0000, and the minimum reporting interval field does not equal 0xffff there SHALL
2707 be no periodic reporting, but change based reporting SHALL still be operational."

min and max are both 0 so it looks like it will be change based reporting (this may be fine?) 

![image](https://github.com/user-attachments/assets/85378b5c-f543-4c79-ba2c-181bcb5c72a8)

Reply:
That was my reasoning too. I wrote `action_report` handling in the Sinope quirk to fire a ZHA event on change, which you wouldn't want to fire periodically, especially when hooked up to an automation.